sdk/storage/storage-queue/src/utils/utils.common.ts

```diff
@@ -115,53 +115,81 @@ export function getURLQueries(url: string): { [key: string]: string } {
 }
 
 /**
- * ONLY AVAILABLE IN NODE.JS RUNTIME.
- *
  * Extracts the parts of an Azure Storage account connection string.
  *
  * @export
  * @param {string} connectionString Connection string.
- * @returns {{ [key: string]: any }} String key value pairs of the storage account's base url for Queue, account name, and account key.
+ * @returns {{ kind: "AccountConnString" | "SASConnString", url: string, [key: string]: any }} String key value pairs of the storage account's url and credentials.
  */
-export function extractConnectionStringParts(connectionString: string): { [key: string]: any } {
-  const matchCredentials = connectionString.match(
-    "DefaultEndpointsProtocol=(.*);AccountName=(.*);AccountKey=(.*);EndpointSuffix=(.*)"
-  );
-
-  let defaultEndpointsProtocol;
-  let accountName;
-  let accountKey;
-  let endpointSuffix;
-
-  try {
-    defaultEndpointsProtocol = matchCredentials![1] || "";
-    accountName = matchCredentials![2] || "";
-    accountKey = Buffer.from(matchCredentials![3], "base64");
-    endpointSuffix = matchCredentials![4] || "";
-  } catch (err) {
-    throw new Error("Invalid Connection String");
-  }
+export function extractConnectionStringParts(
+  connectionString: string
+): { kind: "AccountConnString" | "SASConnString"; url: string; [key: string]: any } {
+  // Account connection string
+  if (
+    connectionString.search("DefaultEndpointsProtocol=") !== -1 &&
+    connectionString.search("AccountKey=") !== -1
+  ) {
+    const matchCredentials = connectionString.match(
```

Comment:
>const matchCredentials = connectionString.match [](start = 4, length = 47)

May not related to this CR about SAS connection string. But seems this kind of connection string cannot be successfully parsed:

```
DefaultEndpointsProtocol=https;
BlobEndpoint=[http://www.mydomain.com](http://www.mydomain.com);
AccountName=storagesample;
AccountKey=<account-key>
```

[https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string#create-a-connection-string-for-an-explicit-storage-endpoint](https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string#create-a-connection-string-for-an-explicit-storage-endpoint)

Reply:
I'm under the impression that connection string can only be of the following type. 
```
DefaultEndpointsProtocol=[http|https];
AccountName=myAccountName;
AccountKey=myAccountKey;
EndpointSuffix=mySuffix;
```
And this is what the portal provides from what I know.

Reply:
We can definitely add support to cover a wider range of connection strings. 
You want me to do it in the same PR or create a new issue for it?

Reply:
Both fine to me. Just what you prefer : )

Reply:
Created a new issue - https://github.com/Azure/azure-sdk-for-js/issues/4578